pyro/distributions/log_normal.py

```diff
@@ -58,13 +45,14 @@ def sample(self, sample_shape=torch.Size()):
         return torch.exp(z)
 
     def batch_log_pdf(self, x):
-        mu = self.mu.expand(self.shape(x))
-        sigma = self.sigma.expand(self.shape(x))
+        mu = self.mu
+        sigma = self.sigma
         ll_1 = Variable(torch.Tensor([-0.5 * np.log(2.0 * np.pi)]).type_as(mu.data).expand_as(x))
         ll_2 = -torch.log(sigma * x)
         ll_3 = -0.5 * torch.pow((torch.log(x) - mu) / sigma, 2.0)
         batch_log_pdf = torch.sum(ll_1 + ll_2 + ll_3, -1)
-        batch_log_pdf_shape = self.batch_shape(x) + (1,)
+        batch_log_pdf_shape = broadcast_shape(self.batch_shape() + self.event_shape(),
```

Comment:
nit: Can you use `self.shape()` in place of `self.batch_shape() + self.event_shape()`? I'm not sure what our semantics is.